addons/calendar/static/src/js/base_calendar.js

```diff
@@ -10,6 +15,154 @@ var WebClient = require('web.WebClient');
+ */
+
+var EventDeleteDialog = Dialog.extend({
+    template: "EventDeleteDialog",
+    events: _.extend({}, Dialog.prototype.events, {
+        'click .o_delete-event': '_onDeleteRecurrentEvents',
+    }),
+    /**
+     * @override
+     * @param {Widget} parent
+     * @param {interger} id calendar event id
+     */
+    init: function (parent, id){
+        //Button cancel that close the modal.
+        var button_cancel = {
+            text: _t("Cancel"),
+            classes: 'btn btn-default',
+            close: true,
+        };
+        this._super(parent, {
+            title: _t('This event is linked to a recurrence.'),
+            size: "medium",
+            buttons: [button_cancel],
+        });
+        this.id = id;
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Detect the action choosen by the user and return it to the CalendarController.
+     *
+     * @param {OdooEvent} event
+     */
+    _onDeleteRecurrentEvents: function (event) {
+        var type = $(event.target).attr('data-type');
+        this.trigger_up('delete_event', {type: type});
+    },
+});
+
+/**
+ * CalendarController
+ *
+ * Add some method in order to handle recurring events.
+ */
+
+CalendarController.include({
+    custom_events: _.extend({}, CalendarController.prototype.custom_events, {
+        delete_event: '_onDeleteRecurrentEvent',
+    }),
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Override the _onDeleteEvent function in oder to change the behavior of calendar popup
+     * delete action. It will insert a new popup in case of recurrent events asking
+     * if the user want to delete only one/future/all event(s).
+     *
+     * @override
+     * @param {integer} id The selected calendar event id
+     */
+    _onDeleteEvent: function (id) {
+        // If is a virtual event, we call the new popup.
+        if (typeof id === "string" && id.indexOf('-') >= 0) {
+            this.dialog.destroy();
+            this.dialog = new EventDeleteDialog(this, id).open();
+        // If a single event, keep the same behavior
+        } else {
+            return this._super.apply(this, arguments);
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * This method call _deleteRecurrentEvents with the type as the action selected by
+     * the user and the calendar event id.
+     * Once the event has been deleted, it will close the modal and refresh the view.
+     *
+     * @param {OdooEvent} event
```

Comment:
@ private

Reply:
@ see _deleteRecurrentEvents

(instead of the message: This method call _deleteRecurrentEvents)